cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/CgmesExportContext.java

```diff
@@ -525,13 +525,36 @@ private void addIidmMappingsGenerators(Network network) {
                 generator.setProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + GENERATING_UNIT, generatingUnit);
             }
             String regulatingControlId = generator.getProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + REGULATING_CONTROL);
-            if (regulatingControlId == null && (generator.isVoltageRegulatorOn() || !Objects.equals(generator, generator.getRegulatingTerminal().getConnectable()))) {
+            if (regulatingControlId == null && canGeneratorControl(generator)) {
                 regulatingControlId = namingStrategy.getCgmesId(ref(generator), Part.REGULATING_CONTROL);
                 generator.setProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + REGULATING_CONTROL, regulatingControlId);
             }
         }
     }
 
+    private static boolean canGeneratorControl(Generator generator) {
+        boolean hasReactiveCapability = false;
+        ReactiveLimits reactiveLimits = generator.getReactiveLimits();
+        if (reactiveLimits != null) {
+            if (reactiveLimits.getKind() == ReactiveLimitsKind.CURVE) {
+                ReactiveCapabilityCurve rcc = (ReactiveCapabilityCurve) reactiveLimits;
+                Collection<ReactiveCapabilityCurve.Point> points = rcc.getPoints();
+                for (ReactiveCapabilityCurve.Point point : points) {
+                    if (point.getMaxQ() != point.getMinQ()) {
+                        hasReactiveCapability = true;
+                        break;
+                    }
+                }
+            } else if (reactiveLimits.getKind() == ReactiveLimitsKind.MIN_MAX) {
+                MinMaxReactiveLimits mmrl = (MinMaxReactiveLimits) reactiveLimits;
+                if (mmrl.getMaxQ() != mmrl.getMinQ()) {
+                    hasReactiveCapability = true;
+                }
+            }
+        }
```

Comment:
As far as I understand, your proposal looks nice, every generator with a correct reactive capability curve will have a regulating control. To be sure: here, there is no more the idea of a difference between local and remote regulating control. Many more regulating control, is it what you expect? Then I wonder about a check on a not Double.NaN targetV.

Reply:
Currently, we're exporting more controls (generators which cannot regulate) so initial assumption was that this will reduce the number of exported controls.

With your suggestion in mind, indeed we might export more controls for generators which are controlling remotely, but I think this is ok, you want to export the true state of the grid. There is no reason not to export remote controls.

`targetValue` is mandatory attribute for `RegulatingControl` so I'll add the check for `Double.NaN` `targetV`.

Reply:
Fixed with the latest commit.